src/arrayMethodJoin.js

```diff
@@ -5,7 +5,41 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let newPhrase = '';
+    let newSeparator = separator;
+
+    switch (separator) {
+      case undefined :
+        newSeparator = ',';
+        break;
+      case null :
+        newSeparator = 'null';
+        break;
+      default:
+        break;
+    }
+
+    for (let i = 0; i < this.length; i++) {
```

Comment:
```suggestion
    for (let i = 0; i < this.length - 1; i++) {
```
і не потрбіна перевірка знизу